state/staking/actions.ts

```diff
@@ -0,0 +1,198 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { monitorTransaction } from 'contexts/RelayerContext';
+import { ThunkConfig } from 'state/types';
+
+import { selectPeriods } from './selectors';
+
+export const fetchStakingData = createAsyncThunk<any, void, ThunkConfig>(
+	'staking/fetchStakingData',
+	async (_, { extra: { sdk } }) => {
+		const {
+			rewardEscrowBalance,
+			stakedNonEscrowedBalance,
+			stakedEscrowedBalance,
+			claimableBalance,
+			kwentaBalance,
+			weekCounter,
+			totalStakedBalance,
+			vKwentaBalance,
+			vKwentaAllowance,
+			kwentaAllowance,
+			epochPeriod,
+			veKwentaBalance,
+			veKwentaAllowance,
+		} = await sdk.kwentaToken.getStakingData();
+
+		return {
+			rewardEscrowBalance: rewardEscrowBalance.toString(),
+			stakedNonEscrowedBalance: stakedNonEscrowedBalance.toString(),
+			stakedEscrowedBalance: stakedEscrowedBalance.toString(),
+			claimableBalance: claimableBalance.toString(),
+			kwentaBalance: kwentaBalance.toString(),
+			weekCounter,
+			totalStakedBalance: totalStakedBalance.toString(),
+			vKwentaBalance: vKwentaBalance.toString(),
+			vKwentaAllowance: vKwentaAllowance.toString(),
+			kwentaAllowance: kwentaAllowance.toString(),
+			epochPeriod,
+			veKwentaBalance: veKwentaBalance.toString(),
+			veKwentaAllowance: veKwentaAllowance.toString(),
+		};
+	}
+);
+
+export const approveKwentaToken = createAsyncThunk<
+	void,
+	'kwenta' | 'vKwenta' | 'veKwenta',
+	ThunkConfig
+>('staking/approveKwentaToken', async (token, { dispatch, extra: { sdk } }) => {
+	const hash = await sdk.kwentaToken.approveKwentaToken(token);
+
+	monitorTransaction({
+		txHash: hash,
+		onTxConfirmed: () => {
+			dispatch(fetchStakingData());
+		},
+	});
+});
+
+export const redeemToken = createAsyncThunk<void, 'vKwenta' | 'veKwenta', ThunkConfig>(
+	'staking/redeemToken',
+	async (token, { dispatch, extra: { sdk } }) => {
+		const hash = await sdk.kwentaToken.redeemToken(token);
+
+		monitorTransaction({
+			txHash: hash,
+			onTxConfirmed: () => {
+				dispatch(fetchStakingData());
+			},
+		});
+	}
+);
+
+export const fetchEscrowData = createAsyncThunk<any, void, ThunkConfig>(
+	'staking/fetchEscrowData',
+	async (_, { extra: { sdk } }) => {
+		const { totalVestable, escrowData } = await sdk.kwentaToken.getEscrowData();
+
+		return { totalVestable, escrowData };
+	}
+);
+
+export const vestEscrowedRewards = createAsyncThunk<any, number[], ThunkConfig>(
```

Comment:
Be good to replace the `any` return types with void if that's all it is, try and avoid `any` where possible.

Reply:
Fixed.